docs/configuration.md

```diff
@@ -114,12 +114,52 @@ in the `spark-defaults.conf` file. A few configuration keys have been renamed si
 versions of Spark; in such cases, the older key names are still accepted, but take lower
 precedence than any instance of the newer key.
 
-Spark properties mainly can be divided into two kinds: one is related to deploy, like
-"spark.driver.memory", "spark.executor.instances", this kind of properties may not be affected when
-setting programmatically through `SparkConf` in runtime, or the behavior is depending on which
-cluster manager and deploy mode you choose, so it would be suggested to set through configuration
-file or `spark-submit` command line options; another is mainly related to Spark runtime control,
-like "spark.task.maxFailures", this kind of properties can be set in either way.
+Note that Spark properties have different effective timing and they can be divided into two kinds:
+<table class="table">
+<tr><th>Configuration Type</th><th>Meaning</th><th>Examples</th></tr>
```

Comment:
Is this title "`Configuration Type`" suitable for `Launching Driver` and `Application Deployment`?

Reply:
`Effect Timing`?

Reply:
eh, `Effective Timing`? cc: @srowen @HyukjinKwon

Reply:
I don't know, "Configuration Time"?

Reply:
Hmm any result? also cc @cloud-fan

Reply:
I'm not good at naming. Basically there are 3 effective timing:
1. when launching the JVM
2. when building the spark context
3. completely dynamic (sql session configs)

Reply:
> I'm not good at naming. Basically there are 3 effective timing:
> 
> 1. when launching the JVM
> 2. when building the spark context
> 3. completely dynamic (sql session configs)

So here should I add back the third type configuration? Since I have remove that section https://github.com/apache/spark/pull/31598#discussion_r584449917

Reply:
Gentle ping @cloud-fan